web/src/ShareSnapshotModal.tsx

```diff
@@ -41,16 +43,17 @@ export default class ShareSnapshotModal extends PureComponent<props> {
       return (
         <section className="ShareSnapshotModal-shareLinkWrap">
           {this.renderShareLink()}
-          {this.renderLoginState()}
+          {this.renderShareLinkInfo()}
+        </section>
+      )
+    } else {
```

Comment:
Oh yeah, I wanted to ask about this — is it more clear to specify if/else (rather than leaving a bare return) to communicate "there are only two possible states that can be rendered"? 

When there's a conditional before a default return, I often take this to mean "you'll always get what's rendered in the return, but sometimes you might additionally get whatever is in the conditional" — but maybe I'm overthinking this?

Reply:
Both communicate "there are only two states" just as well. I think our style has been to eschew else's that return in favor of a bare return at the end of the function, but I don't feel strongly.